src/test/java/org/isf/patient/test/Tests.java

```diff
@@ -232,6 +278,448 @@ public void testMergePatientHistory() throws Exception {
+		testPatient.check(patients.get(patients.size() - 1));
+	}
+
+	@Test
+	public void testMgrGetPatientsPageable() throws Exception {
+		_setupTestPatient(false);
+		ArrayList<Patient> patients = patientBrowserManager.getPatient(0, 10);
+		testPatient.check(patients.get(patients.size() - 1));
+	}
+
+	@Test
+	public void testMgrGetPatientsByOneOfFieldsLike() throws Exception {
+		_setupTestPatient(false);
+		// Pay attention that query return with PAT_ID descendant
+		ArrayList<Patient> patients = patientBrowserManager.getPatientsByOneOfFieldsLike(null);
+		testPatient.check(patients.get(0));
+	}
+
+	@Test
+	public void testMgrGetPatientsByOneOfFieldsLikeFirstName() throws Exception {
+		Integer code = _setupTestPatient(false);
+		Patient foundPatient = patientIoOperation.getPatient(code);
+		ArrayList<Patient> patients = patientBrowserManager.getPatientsByOneOfFieldsLike(foundPatient.getFirstName());
+		testPatient.check(patients.get(0));
+	}
+
+	@Test
+	public void testMgrGetPatientsByOneOfFieldsLikeMiddleOfFirstName() throws Exception {
+		Integer code = _setupTestPatient(false);
+		Patient foundPatient = patientIoOperation.getPatient(code);
+
+		ArrayList<Patient> patients = patientBrowserManager
+				.getPatientsByOneOfFieldsLike(foundPatient.getFirstName().substring(1, foundPatient.getFirstName().length() - 2));
+		testPatient.check(patients.get(0));
+	}
+
+	@Test
+	public void testMgrGetPatientsByOneOfFieldsLikeSecondName() throws Exception {
+		Integer code = _setupTestPatient(false);
+		Patient foundPatient = patientIoOperation.getPatient(code);
+		ArrayList<Patient> patients = patientBrowserManager.getPatientsByOneOfFieldsLike(foundPatient.getSecondName());
+		testPatient.check(patients.get(0));
+	}
+
+	@Test
+	public void testMgrGetPatientsByOneOfFieldsLikeNote() throws Exception {
+		Integer code = _setupTestPatient(false);
+		Patient foundPatient = patientIoOperation.getPatient(code);
+		ArrayList<Patient> patients = patientBrowserManager.getPatientsByOneOfFieldsLike(foundPatient.getSecondName());
+		testPatient.check(patients.get(0));
+	}
+
+	@Test
+	public void testMgrGetPatientsByOneOfFieldsLikeTaxCode() throws Exception {
+		Integer code = _setupTestPatient(false);
+		Patient foundPatient = patientIoOperation.getPatient(code);
+		ArrayList<Patient> patients = patientBrowserManager.getPatientsByOneOfFieldsLike(foundPatient.getTaxCode());
+		testPatient.check(patients.get(0));
+	}
+
+	@Test
+	public void testMgrGetPatientsByOneOfFieldsLikeNotExistingStringShouldNotFindAnything() throws Exception {
+		_setupTestPatient(false);
+		ArrayList<Patient> patients = patientBrowserManager.getPatientsByOneOfFieldsLike("dupa");
+		assertThat(patients).isEmpty();
+	}
+
+	@Test
+	public void testMgrGetPatientByName() throws Exception {
+		Integer code = _setupTestPatient(false);
+		Patient foundPatient = patientIoOperation.getPatient(code);
+		Patient patient = patientBrowserManager.getPatientByName(foundPatient.getName());
```

Comment:
Since i don't know why The method getPatientByName(String) from the type PatientBrowserManager is deprecated, I don't know either why keep using it in testing?

Reply:
My rule is if the method currently exists in the class then write a test for it as someone is probably still using it.   When the method is removed then remove the test.

Reply:
The problem of this method is that searching by name is generally returing a list of results rather than one object only, so it has been deprecated in favor of `getPatientsByOneOfFieldsLike()`.

I think it's ok to remove it soon, along with its test.